### PR TITLE
Fix metadata calculation for breakouts on joined columns 

### LINKED ISSUE #29907: [MLv2] [Bug] Column name calculation is wrong for columns in `:breakout` from joins
Note: uses new mock metadata from #29906

Failing test:

```clj
(deftest ^:parallel metadata-for-join-in-previous-stage-test
  (let [query {:lib/type     :mbql/query
               :lib/metadata meta/metadata-provider
               :database     (meta/id)
               :type         :pipeline
               :stages       [{:lib/type     :mbql.stage/mbql
                               :source-table (meta/id :orders)
                               :joins        [{:lib/type    :mbql/join
                                               :lib/options {:lib/uuid (str (random-uuid))}
                                               :stages      [{:lib/type :mbql.stage/mbql, :source-table (meta/id :products)}]
                                               :alias       "P1"
                                               :conditions   [[:=
                                                               {:lib/uuid (str (random-uuid))}
                                                               [:field
                                                                {:lib/uuid (str (random-uuid))}
                                                                (meta/id :orders :product-id)]
                                                               [:field
                                                                {:join-alias "P1"
                                                                 :lib/uuid   (str (random-uuid))}
                                                                (meta/id :products :id)]]]
                                               :fields      :all}
                                              {:lib/type    :mbql/join
                                               :lib/options {:lib/uuid (str (random-uuid))}
                                               :stages      [{:lib/type :mbql.stage/mbql, :source-table (meta/id :people)}]
                                               :alias       "People"
                                               :conditions   [[:=
                                                               {:lib/uuid (str (random-uuid))}
                                                               [:field
                                                                {:lib/uuid (str (random-uuid))}
                                                                (meta/id :orders :user-id)]
                                                               [:field
                                                                {:join-alias "People"
                                                                 :lib/uuid   (str (random-uuid))}
                                                                (meta/id :people :id)]]]
                                               :fields      :all}]
                               :breakout     [[:field
                                               {:join-alias "P1", :lib/uuid (str (random-uuid))}
                                               (meta/id :products :category)]
                                              [:field
                                               {:join-alias "People", :lib/uuid (str (random-uuid))}
                                               (meta/id :people :source)]]
                               :aggregation  [[:count {:lib/uuid (str (random-uuid))}]]}]}]
    (is (= ["P1__CATEGORY" "People__SOURCE" "count"]
           (mapv :lib/desired-column-alias (lib.metadata.calculation/returned-columns query))))
    (let [query' (update query :stages conj {:lib/type :mbql.stage/mbql
                                             :joins    [{:lib/type    :mbql/join
                                                         :lib/options {:lib/uuid (str (random-uuid))}
                                                         :stages      [{:lib/type     :mbql.stage/mbql
                                                                        :source-table (meta/id :reviews)
                                                                        :joins        [{:lib/type    :mbql/join
                                                                                        :lib/options {:lib/uuid (str (random-uuid))}
                                                                                        :stages      [{:lib/type     :mbql.stage/mbql
                                                                                                       :source-table (meta/id :products)}]
                                                                                        :alias       "P2"
                                                                                        :conditions   [[:=
                                                                                                        {:lib/uuid (str (random-uuid))}
                                                                                                        [:field
                                                                                                         {:lib/uuid (str (random-uuid))}
                                                                                                         (meta/id :reviews :product-id)]
                                                                                                        [:field
                                                                                                         {:join-alias "P2"
                                                                                                          :lib/uuid   (str (random-uuid))}
                                                                                                         (meta/id :products :id)]]]
                                                                                        :fields      :all}]
                                                                        :breakout     [[:field
                                                                                        {:join-alias "P2"
                                                                                         :lib/uuid   (str (random-uuid))}
                                                                                        (meta/id :products :category)]]
                                                                        :aggregation  [[:avg
                                                                                        {:lib/uuid (str (random-uuid))}
                                                                                        [:field
                                                                                         {:lib/uuid (str (random-uuid))}
                                                                                         (meta/id :reviews :rating)]]]}
                                                                       {:lib/type :mbql.stage/mbql}]
                                                         :alias       "Q2"
                                                         :conditions   [[:=
                                                                         {:lib/uuid (str (random-uuid))}
                                                                         [:field
                                                                          {:join-alias "P1", :lib/uuid (str (random-uuid))}
                                                                          (meta/id :products :category)]
                                                                         [:field
                                                                          {:join-alias "Q2", :lib/uuid (str (random-uuid))}
                                                                          (meta/id :products :category)]]]
                                                         :fields      :all}]
                                             :order-by [[:asc
                                                         {:lib/uuid (str (random-uuid))}
                                                         [:field
                                                          {:join-alias "P1", :lib/uuid (str (random-uuid))}
                                                          (meta/id :products :category)]]
                                                        [:asc
                                                         {:lib/uuid (str (random-uuid))}
                                                         [:field
                                                          {:join-alias "People", :lib/uuid (str (random-uuid))}
                                                          (meta/id :people :source)]]]
                                             :limit    2})]
      (is (= ["P1__CATEGORY" "People__SOURCE" "count" "Q2__P2__CATEGORY" "Q2__avg"]
             (mapv :lib/desired-column-alias (lib.metadata.calculation/returned-columns query')))))))
```

Failures:

```clj
Fail in metadata-for-join-in-previous-stage-test

expected: ["P1__CATEGORY" "People__SOURCE" "count"]

  actual: ["CATEGORY" "SOURCE" "count"]          
    diff: - ["P1__CATEGORY" "People__SOURCE"]          
          + ["CATEGORY" "SOURCE"]            


Fail in metadata-for-join-in-previous-stage-test

expected: ["P1__CATEGORY" "People__SOURCE" "count" "Q2__P2__CATEGORY" "Q2__avg"]            

  actual: ["CATEGORY" "SOURCE" "count" "Q2__CATEGORY" "Q2__avg"]          
    diff: - ["P1__CATEGORY" "People__SOURCE" nil "Q2__P2__CATEGORY"]          
          + ["CATEGORY" "SOURCE" nil "Q2__CATEGORY"]            
```

### PR DESCRIPTION
**Pull request by camsaul**
The `:lib/source-column-alias` and `:lib/desired-column-alias` calculation logic for breakouts was wrong; it was always using `:name` rather than propagating previous source/desired column aliases. This meant breakouts that use columns from joins could potentially break. 

Fixes #29907

### CODE REVIEW

File: src/metabase/lib/join.cljc [hunk 1]
```diff
@@ -287,11 +289,11 @@
   "Create an MBQL join map from something that can conceptually be joined against. A `Table`? An MBQL or native query? A
   Saved Question? You should be able to join anything, and this should return a sensible MBQL join map."
   ([joinable]
-   (join-clause-method joinable))
+   (-> (join-clause-method joinable)
+       (u/assoc-default :fields :all)))
```
Comment (camsaul) [inline src/metabase/lib/join.cljc:293]: Unrelated fix, `:fields :all` wasn't being added to join clauses by default if you called the 1 arity

File: test/metabase/lib/join_test.cljc [hunk 2]
```diff
@@ -55,6 +55,16 @@
                                              :args     [(lib/ref (meta/field-metadata :venues :category-id))
                                                         (lib/ref (meta/field-metadata :categories :id))]}]))))))
 
+(deftest ^:parallel join-clause-test
```
Comment (camsaul) [inline test/metabase/lib/join_test.cljc:58]: test for unrelated fix

File: test/metabase/query_processor/middleware/constraints_test.clj [hunk 3]
```diff
@@ -7,26 +7,26 @@
 (defn- add-default-userland-constraints [query]
   (:pre (mt/test-qp-middleware qp.constraints/add-default-userland-constraints query)))
 
-(deftest no-op-without-middleware-options-test
+(deftest ^:parallel no-op-without-middleware-options-test
```
Comment (camsaul) [inline test/metabase/query_processor/middleware/constraints_test.clj:10]: unrelated but I noticed these are all pure tests and can be marked parallel

File: test/metabase/test_runner.clj [hunk 4]
```diff
@@ -84,7 +84,7 @@
    "test_resources"])
 
 (defn- default-options []
-  {:namespace-pattern   #"^metabase.*test$"
+  {:namespace-pattern   #"^metabase.*"
```
Comment (camsaul) [inline test/metabase/test_runner.clj:87]: Also unrelated, we have a few random tests in places like `metabase.test-util` and our test runner shouldn't skip these tests just because the namespace doesn't end in `test`. Look for tests in all `^metabase` namespaces.

File: test/metabase/actions/test_util.clj [hunk 5]
```diff
@@ -124,7 +145,7 @@
             (testing "delete row"
               (is (= [1]
                      (jdbc/execute! (sql-jdbc.conn/db->pooled-connection-spec (data/id))
-                                    "DELETE FROM CATEGORIES WHERE ID = 1;"))))
+                                    (delete-categories-1-query)))))
```
Comment (camsaul) [inline test/metabase/actions/test_util.clj:148]: This test previously wasn't running, because this namespace didn't end in `-test`; now that it's enabled, it was breaking in MySQL because

```
 delete row
expected: (=
           [1]
           (jdbc/execute!
            (sql-jdbc.conn/db->pooled-connection-spec (data/id))
            "DELETE FROM CATEGORIES WHERE ID = 1;"))
  actual: 
  java.sql.SQLSyntaxErrorException: (conn=226) Table 'actions-test-data.CATEGORIES' doesn't exist
```

So it never actually worked in the first place for MySQL. Fixed by making sure we generate a quoted query appropriate for the current driver

File: modules/drivers/presto-jdbc/test/metabase/test/data/presto_jdbc.clj [hunk 6]
```diff
@@ -163,7 +163,7 @@
   (testing "Make sure logic to strip out NOT NULL and PRIMARY KEY stuff works as expected"
     (let [db-def    (tx/get-dataset-definition defs/test-data)
           table-def (-> db-def :table-definitions second)]
-      (is (= "CREATE TABLE \"test_data\".\"default\".\"categories\" (\"id\" INTEGER, \"name\" VARCHAR) ;"
+      (is (= "CREATE TABLE \"test_data\".\"default\".\"test_data_categories\" (\"id\" INTEGER, \"name\" VARCHAR) ;"
```
Comment (camsaul) [inline modules/drivers/presto-jdbc/test/metabase/test/data/presto_jdbc.clj:166]: This test was also not running and was also broken. Fixed

File: src/metabase/lib/join.cljc (hunk 1 above)
Comment (bshepherdson) [reply src/metabase/lib/join.cljc:293]: It looks now like it might not be called for with the 2-arity?

**Review by bshepherdson — APPROVED** [review]

File: src/metabase/lib/join.cljc (hunk 1 above)
Comment (camsaul) [reply src/metabase/lib/join.cljc:293]: The two-arity calls the one-arity
